### ISSUE TITLE
Add setting to `import/named` to disabled error on specific identifiers

### ISSUE THREAD

**Issue by taylor1791** [benmosher/eslint-plugin-import#1215]
It is not uncommon to import something that is provided by a babel-plugin. One such example is [babel-plugin-rewire](https://github.com/speedskater/babel-plugin-rewire). It provides `__Rewire__` and `__ResetDependency__` to "override" local variables.

I purpose adding options to `import/named` to disable the the error on specific identifiers. Here is an example.

```
rules: {
  'named/import': [2, {allowedNames: ["__Rewire__", "__RestDependency__"]}]
}
```

If approved, I can make an MR.

**Comment by ljharb** [comment]
I'm not sure I understand what you mean from reading the linked example. Can you elaborate?

**Comment by taylor1791** [comment]
Let say you have the following contrived example

env.js
```js
var env = 'production';

export default function(name) {
  return env;
}
```

test.js
```js
import getEnv, {__Rewire__, __ResetDependency__} from './env';
__Rewire__('env', 'testing');

if (getEnv() !== 'testing') throw new Error("Incorrect environment");
```

Rightfully so, `import/named` will complain about `__Rewire__` and `__ResetDependency__` not being exported.

**Comment by ljharb** [comment]
hmm, that seems like a really really strange pattern. Is there any usage of it in the ecosystem besides with rewire?

**Comment by taylor1791** [comment]
I agree it is a weird pattern.

**Comment by taylor1791** [comment]
I agree it is weird and we are trying to move away from it. `babel-plugin-rewire` is the one I am most familiar with. `babel-plugin-rewire` is more popular than I anticipated with 50k downloaded this week.

I think a feature like this could also be useful if the exports were generated at build time using something like [babel-plugin-preval](https://github.com/kentcdodds/babel-plugin-preval) it only has 4k downloads this week.

Both of these are mentioned on [awesome-babel](https://github.com/babel/awesome-babel)

If I really try to stretch it, I could imagine people using webpack's `alias` to "create" importable packages. These aliases wouldn't work on units test runners. However they could use NODE_PATH when testing to point to a test mock during tests.

I might be able to do a more exhaustive search of the community if you think you more examples would help.
